File: src/main/kotlin/cl/ravenhill/pijamaparty/againstallodds/EvenOrOdd.java
package cl.ravenhill.pijamaparty.againstallodds;

public class EvenOrOdd {
  public static void main(String[] args) {
    EvenOrOdd evenOrOdd = new EvenOrOdd();
    System.out.println(evenOrOdd.isEven(2));
    System.out.println(evenOrOdd.isOdd(1));
    System.out.println(evenOrOdd.isEven(1));
  }

  public boolean isEven(int n) {
    return n >= 0 && (n == 0 || isOdd(n - 1));
  }

  public boolean isOdd(int n) {
    return n > 0 && isEven(n - 1);
  }
}

File: src/main/kotlin/cl/ravenhill/pijamaparty/againstallodds/EvenOrOdd.kt
package cl.ravenhill.pijamaparty.againstallodds

fun main() {
  println(isEven(2))
  println(isOdd(5))
  println(isEven(7))
}

fun isEven(n: Int) = n >= 0 && (n == 0 || isOdd(n - 1))

fun isOdd(n: Int): Boolean = n > 0 && isEven(n - 1)
